### ISSUE TITLE
[analyzer] Convert to expression body refactoring with IIFE produces invalid programs.

### ISSUE THREAD

**Issue by modulovalue** [dart-lang/sdk#51421]
Consider the following:

```dart
void main() {
  final a = () {
    return 0;
  }();
}
```

If we use the 'Convert to expression body' refactoring to convert the body of the function literal into an expression body, we will end up with an invalid program:

```dart
void main() {
  final a = () => 0();
}
```

I think that we should probably remove the function invocation and the surrounding function expression if this refactoring is invoked on an IIFE containing a single return statement:

```dart
void main() {
  final a = 0;
}
```

**Comment by asashour** [comment]
https://dart-review.googlesource.com/c/sdk/+/283360

**Comment by modulovalue** [comment]
@asashour I'm wondering what happened to https://dart-review.googlesource.com/c/sdk/+/283360 it looks like it is done but was not merged?

**Comment by asashour** [comment]
I am under the impression that the team is busy for other issues, needed for the beta release. Kind reminders would be send after it is shipped.

**Comment by bwilkerson** [comment]
That is correct. We don't currently have many cycles available for work that isn't directly related to Dart 3.0

**Comment by modulovalue** [comment]
I see, that makes sense to me. Thank you for the explanation!

**Comment by FMorschel** [comment]
@asashour maybe rebasing it now would make it possible for them to merge? WDYT?
